File: spring-integration/basic-integration/src/main/java/com/ddubson/integration/PrintService.java
package com.ddubson.integration;

import com.ddubson.LogAdapter;
import org.springframework.messaging.Message;
import org.springframework.messaging.support.MessageBuilder;

import static com.ddubson.ANSIColor.*;

public class PrintService {
    private final LogAdapter logAdapter;

    public PrintService(LogAdapter logAdapter) {
        this.logAdapter = logAdapter;
    }

    public Message<?> print(Message<String> msg) {
        msg.getHeaders().forEach((k, v) -> logAdapter.info("[2] Header: " + k + ":" + v, ANSI_GREEN));
        logAdapter.info("[2] Payload: " + msg.getPayload(), ANSI_GREEN);

        return MessageBuilder.withPayload(msg.getHeaders().get("id")).build();
    }
}

File: spring-integration/basic-integration/src/main/kotlin/com/ddubson/integration/PrintService.kt
package com.ddubson.integration

import com.ddubson.ANSIColor.*
import com.ddubson.LogAdapter
import org.springframework.messaging.Message
import org.springframework.messaging.support.MessageBuilder

class PrintService(val logAdapter: LogAdapter) {
    fun print(msg: Message<String>): Message<Any?>? {
        msg.headers.forEach { (k, v) ->
            logAdapter.info("[2] Header: $k:$v", ANSI_GREEN)
        }

        logAdapter.info("[2] Payload: ${msg.payload}", ANSI_GREEN)
        return MessageBuilder.withPayload(msg.headers["id"]).build()
    }
}